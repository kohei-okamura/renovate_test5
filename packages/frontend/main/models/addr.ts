/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Prefecture } from '@zinger/enums/lib/prefecture'

/**
 * 住所.
 */
export type Addr = Readonly<{
  /** 郵便番号 */
  postcode: string

  /** 都道府県 */
  prefecture: Prefecture

  /** 市区町村 */
  city: string

  /** 町名・番地 */
  street: string

  /** 建物名など */
  apartment: string
}>
