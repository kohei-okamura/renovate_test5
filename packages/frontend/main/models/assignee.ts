/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */

import { StaffId } from '~/models/staff'

/**
 * 予実担当スタッフ.
 */
export type Assignee = Readonly<{
  /** スタッフID */
  staffId: StaffId | undefined

  /** 未定フラグ */
  isUndecided: boolean

  /** 研修フラグ */
  isTraining: boolean
}>
