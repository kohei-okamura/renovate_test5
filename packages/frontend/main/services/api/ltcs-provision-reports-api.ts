/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import {
  HomeVisitLongTermCareSpecifiedOfficeAddition
} from '@zinger/enums/lib/home-visit-long-term-care-specified-office-addition'
import { LtcsBaseIncreaseSupportAddition } from '@zinger/enums/lib/ltcs-base-increase-support-addition'
import { LtcsOfficeLocationAddition } from '@zinger/enums/lib/ltcs-office-location-addition'
import { LtcsProvisionReportStatus } from '@zinger/enums/lib/ltcs-provision-report-status'
import {
  LtcsSpecifiedTreatmentImprovementAddition
} from '@zinger/enums/lib/ltcs-specified-treatment-improvement-addition'
import { LtcsTreatmentImprovementAddition } from '@zinger/enums/lib/ltcs-treatment-improvement-addition'
import { AxiosInstance } from 'axios'
import { DateLike } from '~/models/date'
import { Job } from '~/models/job'
import { LtcsProvisionReport } from '~/models/ltcs-provision-report'
import { LtcsProvisionReportDigest } from '~/models/ltcs-provision-report-digest'
import { LtcsProvisionReportEntry } from '~/models/ltcs-provision-report-entry'
import { LtcsProvisionReportOverScore } from '~/models/ltcs-provision-report-over-score'
import { OfficeId } from '~/models/office'
import { UserId } from '~/models/user'
import { api, Api } from '~/services/api/core'

/**
 * 介護保険サービス予実 API.
 */
export namespace LtcsProvisionReportsApi {
  type Identifiers = {
    officeId: OfficeId
    userId: UserId
    providedIn: DateLike
  }

  export type DeleteParams = Identifiers

  export type BatchResponse = {
    job: Job
  }

  export type DownloadForm = {
    officeId: OfficeId
    userId: UserId
    providedIn: DateLike
    issuedOn: DateLike
    needsMaskingInsNumber?: boolean | ''
    needsMaskingInsName?: boolean | ''
  }
  export type DownloadParams = {
    form: DownloadForm
  }
  export type Download = {
    downloadSheets (params: DownloadParams): Promise<BatchResponse>
  }

  export type GetParams = Identifiers
  export type GetResponse = {
    ltcsProvisionReport: LtcsProvisionReport
  }

  export type GetIndexParams = Api.GetIndexParams & {
    officeId?: number | ''
    providedIn?: string
    status?: LtcsProvisionReportStatus | ''
    q?: string
  }
  export type GetIndexResponse = Api.GetIndexResponse<LtcsProvisionReportDigest>

  export type GetScoreSummaryForm = Identifiers & {
    entries: LtcsProvisionReportEntry[]
    specifiedOfficeAddition: HomeVisitLongTermCareSpecifiedOfficeAddition
    treatmentImprovementAddition: LtcsTreatmentImprovementAddition
    specifiedTreatmentImprovementAddition: LtcsSpecifiedTreatmentImprovementAddition
    baseIncreaseSupportAddition: LtcsBaseIncreaseSupportAddition
    locationAddition: LtcsOfficeLocationAddition
    plan: LtcsProvisionReportOverScore
    result: LtcsProvisionReportOverScore
  }
  export type GetScoreSummaryParams = Api.Form<GetScoreSummaryForm>
  export type GetScoreSummaryResponse = {
    plan: {
      managedScore: number
      unmanagedScore: number
    }
    result: {
      managedScore: number
      unmanagedScore: number
    }
  }
  export type GetScoreSummary = {
    getScoreSummary (params: GetScoreSummaryParams): Promise<GetScoreSummaryResponse>
  }

  export type UpdateForm = {
    entries: LtcsProvisionReportEntry[]
    specifiedOfficeAddition: HomeVisitLongTermCareSpecifiedOfficeAddition
    treatmentImprovementAddition: LtcsTreatmentImprovementAddition
    specifiedTreatmentImprovementAddition: LtcsSpecifiedTreatmentImprovementAddition
    baseIncreaseSupportAddition: LtcsBaseIncreaseSupportAddition
    locationAddition: LtcsOfficeLocationAddition
    plan: LtcsProvisionReportOverScore
    result: LtcsProvisionReportOverScore
  }
  export type UpdateParams = Api.Form<UpdateForm> & Identifiers
  export type UpdateResponse = GetResponse

  export type UpdateStatusForm = {
    status: LtcsProvisionReportStatus
  }
  export type UpdateStatusParams = Api.Form<UpdateStatusForm> & Identifiers
  export type UpdateStatusResponse = GetResponse
  export type UpdateStatus = {
    updateStatus (params: UpdateStatusParams): Promise<UpdateStatusResponse>
  }

  export type Definition =
    Api.Delete<DeleteParams> &
    Api.Get<GetResponse, GetParams> &
    Api.GetIndex<GetIndexParams, GetIndexResponse> &
    Api.Update<UpdateForm, UpdateParams, UpdateResponse> &
    Download &
    GetScoreSummary &
    UpdateStatus

  const endpoint = (params?: GetParams, last?: string) => {
    const segments = params ? [params.officeId, params.userId, `${params.providedIn}`, last] : []
    return api.endpoint('ltcs-provision-reports', ...segments)
  }

  export const create = (axios: AxiosInstance): Definition => ({
    async delete (params) {
      await axios.delete(endpoint(params))
    },
    // サービス提供票
    async downloadSheets ({ form }) {
      return await api.extract(axios.post(api.endpoint('ltcs-provision-report-sheets'), form))
    },
    async get (params) {
      return await api.extract(axios.get(endpoint(params)))
    },
    async getIndex (params) {
      return await api.extract(axios.get(endpoint(), { params }))
    },
    async getScoreSummary (params) {
      return await api.extract(axios.post(api.endpoint('ltcs-provision-report-score-summary'), params.form))
    },
    async update (params) {
      return await api.extract(axios.put(endpoint(params), params.form))
    },
    async updateStatus (params) {
      return await api.extract(axios.put(endpoint(params, 'status'), params.form))
    }
  })
}
