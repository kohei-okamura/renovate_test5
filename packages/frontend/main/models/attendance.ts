/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ServiceOption } from '@zinger/enums/lib/service-option'
import { Task } from '@zinger/enums/lib/task'
import { Assignee } from '~/models/assignee'
import { ContractId } from '~/models/contract'
import { DateLike } from '~/models/date'
import { Duration } from '~/models/duration'
import { OfficeId } from '~/models/office'
import { Schedule } from '~/models/schedule'
import { ShiftId } from '~/models/shift'
import { StaffId } from '~/models/staff'
import { UserId } from '~/models/user'

/**
 * 勤務実績 ID.
 */
export type AttendanceId = number

/**
 * 勤務実績.
 */
export type Attendance = Readonly<{
  /** 勤務実績 ID */
  id: AttendanceId

  /** 勤務シフト ID */
  shiftId: ShiftId | undefined

  /** 契約 ID */
  contractId: ContractId | undefined

  /** 事業所 ID */
  officeId: OfficeId

  /** 利用者 ID */
  userId: UserId | undefined

  /** 管理スタッフ ID */
  assignerId: StaffId

  /** 予実区分 */
  task: Task

  /** サービスコード */
  serviceCode: string

  /** 頭数 */
  headcount: number

  /** 担当スタッフ */
  assignees: Assignee[]

  /** スケジュール */
  schedule: Schedule

  /** 予実所要時間 */
  durations: Duration[]

  /** オプション */
  options: ServiceOption[]

  /** 備考 */
  note: string

  /** 確定フラグ */
  isConfirmed: boolean

  /** キャンセルフラグ */
  isCanceled: boolean

  /** キャンセル理由 */
  reason: string | undefined

  /** 登録日時 */
  createdAt: DateLike

  /** 更新日時 */
  updatedAt: DateLike
}>
