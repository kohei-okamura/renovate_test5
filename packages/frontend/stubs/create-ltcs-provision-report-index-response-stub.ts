/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { LtcsProvisionReportsApi } from '~/services/api/ltcs-provision-reports-api'
import { createIndexResponse } from '~~/stubs/create-index-response'
import {
  createLtcsProvisionReportDigestStubs,
  LTCS_PROVISION_REPORT_DIGEST_STUB_COUNT
} from '~~/stubs/create-ltcs-provision-report-digest-stub'

export function createLtcsProvisionReportIndexResponseStub (
  params: LtcsProvisionReportsApi.GetIndexParams = {}
): LtcsProvisionReportsApi.GetIndexResponse {
  return createIndexResponse(params, LTCS_PROVISION_REPORT_DIGEST_STUB_COUNT, createLtcsProvisionReportDigestStubs)
}
