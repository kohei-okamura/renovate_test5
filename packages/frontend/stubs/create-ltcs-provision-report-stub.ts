/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import {
  HomeVisitLongTermCareSpecifiedOfficeAddition
} from '@zinger/enums/lib/home-visit-long-term-care-specified-office-addition'
import { LtcsBaseIncreaseSupportAddition } from '@zinger/enums/lib/ltcs-base-increase-support-addition'
import { LtcsOfficeLocationAddition } from '@zinger/enums/lib/ltcs-office-location-addition'
import { LtcsProvisionReportStatus } from '@zinger/enums/lib/ltcs-provision-report-status'
import {
  LtcsSpecifiedTreatmentImprovementAddition
} from '@zinger/enums/lib/ltcs-specified-treatment-improvement-addition'
import { LtcsTreatmentImprovementAddition } from '@zinger/enums/lib/ltcs-treatment-improvement-addition'
import { LtcsProvisionReport } from '~/models/ltcs-provision-report'
import { CONTRACT_ID_MAX, CONTRACT_ID_MIN } from '~~/stubs/create-contract-stub'
import { createLtcsProvisionReportEntryStubs } from '~~/stubs/create-ltcs-provision-report-entry-stub'
import { OFFICE_ID_MAX, OFFICE_ID_MIN } from '~~/stubs/create-office-stub'
import { USER_ID_MAX, USER_ID_MIN } from '~~/stubs/create-user-stub'
import { createFaker } from '~~/stubs/fake'
import { ID_MAX, ID_MIN, SEEDS } from '~~/stubs/index'

export const LTCS_PROVISION_REPORT_ID_MAX = ID_MAX
export const LTCS_PROVISION_REPORT_ID_MIN = ID_MIN

type Arguments = {
  id?: number
  providedIn?: string
}

export function createLtcsProvisionReportStub (args: Arguments = {}): LtcsProvisionReport {
  const id = args.id ?? LTCS_PROVISION_REPORT_ID_MIN
  const faker = createFaker(SEEDS[id - 1])
  const providedIn = args?.providedIn ?? '2021-02'
  const status = faker.randomElement(LtcsProvisionReportStatus.values)
  return {
    /** 予実 ID */
    id,
    /** 利用者 ID */
    userId: faker.intBetween(USER_ID_MIN, USER_ID_MAX),
    /** 事業所 ID */
    officeId: faker.intBetween(OFFICE_ID_MIN, OFFICE_ID_MAX),
    /** 契約 ID */
    contractId: faker.intBetween(CONTRACT_ID_MIN, CONTRACT_ID_MAX),
    /** サービス提供年月 */
    providedIn,
    /** サービス情報 */
    entries: createLtcsProvisionReportEntryStubs(providedIn),
    /** 特定事業所加算 */
    specifiedOfficeAddition: faker.randomElement(HomeVisitLongTermCareSpecifiedOfficeAddition.values),
    /** 処遇改善加算 */
    treatmentImprovementAddition: faker.randomElement(LtcsTreatmentImprovementAddition.values),
    /** 特定処遇改善加算 */
    specifiedTreatmentImprovementAddition: faker.randomElement(LtcsSpecifiedTreatmentImprovementAddition.values),
    /** ベースアップ等支援加算 */
    baseIncreaseSupportAddition: faker.randomElement(LtcsBaseIncreaseSupportAddition.values),
    /** 地域加算 */
    locationAddition: faker.randomElement(LtcsOfficeLocationAddition.values),
    /** 超過単位（予定） */
    plan: {
      maxBenefitQuotaExcessScore: faker.intBetween(10, 99999),
      maxBenefitExcessScore: faker.intBetween(10, 99999)
    },
    /** 超過単位（実績） */
    result: {
      maxBenefitQuotaExcessScore: faker.intBetween(10, 99999),
      maxBenefitExcessScore: faker.intBetween(10, 99999)
    },
    /** 状態 */
    status,
    /** 確定日時 */
    fixedAt: status === LtcsProvisionReportStatus.fixed ? faker.randomDateString() : undefined,
    /** 登録日時 */
    createdAt: faker.randomDateString(),
    /** 更新日時 */
    updatedAt: faker.randomDateString()
  }
}
