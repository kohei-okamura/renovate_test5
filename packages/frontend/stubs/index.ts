/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createFaker } from '~~/stubs/fake'

export const ID_MAX = 100
export const ID_MIN = 1
export const STUB_COUNT = 1000
export const STUB_DEFAULT_SEED = 'zinger'

export const createSeeds = (n: number, seed: string = STUB_DEFAULT_SEED): string[] => {
  const faker = createFaker(seed)
  return Array.from(Array(n), () => faker.randomString(8))
}

export const SEEDS = createSeeds(STUB_COUNT * 3)

export type CreateStubs<T> = {
  (n?: number, skip?: number): T[]
}
