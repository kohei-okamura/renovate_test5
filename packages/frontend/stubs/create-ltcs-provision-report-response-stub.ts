/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { LtcsProvisionReportsApi } from '~/services/api/ltcs-provision-reports-api'
import { createLtcsProvisionReportStub } from '~~/stubs/create-ltcs-provision-report-stub'

type Arguments = {
  id?: number
  providedIn?: string
}

export function createLtcsProvisionReportResponseStub (args: Arguments = {}): LtcsProvisionReportsApi.GetResponse {
  return {
    ltcsProvisionReport: createLtcsProvisionReportStub(args)
  }
}
