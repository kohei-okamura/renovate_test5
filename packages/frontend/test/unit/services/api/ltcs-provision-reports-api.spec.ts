/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { HomeHelpServiceSpecifiedOfficeAddition } from '@zinger/enums/lib/home-help-service-specified-office-addition'
import { LtcsBaseIncreaseSupportAddition } from '@zinger/enums/lib/ltcs-base-increase-support-addition'
import { LtcsOfficeLocationAddition } from '@zinger/enums/lib/ltcs-office-location-addition'
import { LtcsProvisionReportStatus } from '@zinger/enums/lib/ltcs-provision-report-status'
import {
  LtcsSpecifiedTreatmentImprovementAddition
} from '@zinger/enums/lib/ltcs-specified-treatment-improvement-addition'
import { LtcsTreatmentImprovementAddition } from '@zinger/enums/lib/ltcs-treatment-improvement-addition'
import Axios from 'axios'
import { HttpStatusCode } from '~/models/http-status-code'
import { LtcsProvisionReportsApi } from '~/services/api/ltcs-provision-reports-api'
import { createLtcsProvisionReportIndexResponseStub } from '~~/stubs/create-ltcs-provision-report-index-response-stub'
import { createLtcsProvisionReportResponseStub } from '~~/stubs/create-ltcs-provision-report-response-stub'
import { createLtcsProvisionReportStub } from '~~/stubs/create-ltcs-provision-report-stub'
import { createAxiosMockAdapter } from '~~/test/helpers/create-axios-mock-adapter'
import { deleteUndefinedProperties } from '~~/test/helpers/delete-undefined-properties'

describe('api/ltcs-provision-reports-api', () => {
  const axios = Axios.create()
  const adapter = createAxiosMockAdapter(axios)
  const baseUrl = '/api/ltcs-provision-reports'
  let ltcsProvisionReports: LtcsProvisionReportsApi.Definition

  const createUrl = ({ officeId, userId, providedIn }: LtcsProvisionReportsApi.GetParams) => {
    return `${baseUrl}/${officeId}/${userId}/${providedIn}`
  }

  beforeEach(() => {
    ltcsProvisionReports = LtcsProvisionReportsApi.create(axios)
  })

  describe('getIndex', () => {
    const params: LtcsProvisionReportsApi.GetIndexParams = {
      officeId: 1,
      providedIn: '2021-04',
      status: LtcsProvisionReportStatus.fixed,
      q: 'キーワード'
    }

    it(`should get ${baseUrl}`, async () => {
      adapter.setup(x => {
        x.onGet(baseUrl).replyOnce(HttpStatusCode.OK, createLtcsProvisionReportIndexResponseStub(params))
      })

      await ltcsProvisionReports.getIndex(params)

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'get', url: baseUrl })
    })

    it('should return response of the api', async () => {
      adapter.setup(x => {
        x.onGet(baseUrl).replyOnce(HttpStatusCode.OK, createLtcsProvisionReportIndexResponseStub(params))
      })

      const result = await ltcsProvisionReports.getIndex(params)

      expect(result).toMatchSnapshot()
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onGet(baseUrl).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = ltcsProvisionReports.getIndex(params)

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe('get', () => {
    const params = {
      officeId: 1,
      userId: 10,
      providedIn: '2021-01'
    }
    const url = createUrl(params)

    it(`should get ${baseUrl}/:officeId/:userId/:providedIn`, async () => {
      adapter.setup(x => {
        x.onGet(url).replyOnce(HttpStatusCode.OK, createLtcsProvisionReportResponseStub())
      })

      await ltcsProvisionReports.get(params)

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'get', url })
    })

    it('should return response of the api', async () => {
      const expected = createLtcsProvisionReportResponseStub()

      adapter.setup(x => {
        x.onGet(url).replyOnce(HttpStatusCode.OK, expected)
      })

      const response = await ltcsProvisionReports.get(params)

      expect(response).toStrictEqual(deleteUndefinedProperties(expected))
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onGet(url).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = ltcsProvisionReports.get(params)

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe('update', () => {
    const params = {
      officeId: 1,
      userId: 10,
      providedIn: '2021-01'
    }
    const url = createUrl(params)
    const current = createLtcsProvisionReportResponseStub()
    const ltcsProvisionReport = createLtcsProvisionReportStub({ id: current.ltcsProvisionReport.id })
    const expected = { ltcsProvisionReport }
    const form: LtcsProvisionReportsApi.UpdateForm = {
      entries: ltcsProvisionReport.entries,
      specifiedOfficeAddition: HomeHelpServiceSpecifiedOfficeAddition.addition4,
      treatmentImprovementAddition: LtcsTreatmentImprovementAddition.none,
      specifiedTreatmentImprovementAddition: LtcsSpecifiedTreatmentImprovementAddition.addition2,
      baseIncreaseSupportAddition: LtcsBaseIncreaseSupportAddition.addition1,
      locationAddition: LtcsOfficeLocationAddition.mountainousArea,
      plan: ltcsProvisionReport.plan,
      result: ltcsProvisionReport.result
    }

    it(`should put ${baseUrl}/:officeId/:userId/:providedIn`, async () => {
      adapter.setup(x => x.onPut(url).replyOnce(HttpStatusCode.NoContent))

      await ltcsProvisionReports.update({ ...params, form })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'put', url })
      expect(JSON.parse(request.data)).toStrictEqual(deleteUndefinedProperties(form))
    })

    it('should return response of the api', async () => {
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.OK, expected)
      })

      const response = await ltcsProvisionReports.update({ ...params, form })

      expect(response).toStrictEqual(deleteUndefinedProperties(expected))
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = ltcsProvisionReports.update({ ...params, form })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe('updateStatus', () => {
    const officeId = 2
    const userId = 30
    const providedIn = '2021-04-05'
    const params = {
      officeId,
      userId,
      providedIn
    }
    const url = `${baseUrl}/${officeId}/${userId}/${providedIn}/status`
    const form: LtcsProvisionReportsApi.UpdateStatusForm = {
      status: LtcsProvisionReportStatus.fixed
    }

    it('should put /api/ltcs-provision-reports/:officeId/:userId/:providedIn/status', async () => {
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.OK, createLtcsProvisionReportResponseStub({ providedIn }))
      })

      await ltcsProvisionReports.updateStatus({ ...params, form })

      const actual = adapter.getLastRequest()
      expect(actual).toMatchObject({ method: 'put', url })
      expect(JSON.parse(actual.data)).toStrictEqual(form)
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.BadRequest)
      })

      const actual = ltcsProvisionReports.updateStatus({ ...params, form })

      await expect(actual).rejects.toThrowErrorMatchingSnapshot()
    })

    it('should return response of the api', async () => {
      const response = createLtcsProvisionReportResponseStub({ providedIn })
      const expected = deleteUndefinedProperties(response)
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.OK, response)
      })

      const actual = await ltcsProvisionReports.updateStatus({ ...params, form })

      expect(actual).toStrictEqual(expected)
    })
  })

  describe('ltcs-provision-report-score-summary', () => {
    const url = '/api/ltcs-provision-report-score-summary'
    const stub = createLtcsProvisionReportResponseStub()
    const ltcsProvisionReport = createLtcsProvisionReportStub({ id: stub.ltcsProvisionReport.id })
    const form: LtcsProvisionReportsApi.GetScoreSummaryForm = {
      officeId: 1,
      userId: 10,
      providedIn: '2021-01',
      entries: ltcsProvisionReport.entries,
      specifiedOfficeAddition: HomeHelpServiceSpecifiedOfficeAddition.none,
      treatmentImprovementAddition: LtcsTreatmentImprovementAddition.addition1,
      specifiedTreatmentImprovementAddition: LtcsSpecifiedTreatmentImprovementAddition.none,
      baseIncreaseSupportAddition: LtcsBaseIncreaseSupportAddition.none,
      locationAddition: LtcsOfficeLocationAddition.none,
      plan: ltcsProvisionReport.plan,
      result: ltcsProvisionReport.result
    }

    it('should post ltcs-provision-report-score-summary', async () => {
      adapter.setup(x => x.onPost(url).replyOnce(HttpStatusCode.NoContent))

      await ltcsProvisionReports.getScoreSummary({ form })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'post', url })
      expect(JSON.parse(request.data)).toStrictEqual(deleteUndefinedProperties(form))
    })

    it('should return response of the api', async () => {
      const expected = { plan: 10000, result: 10000 }

      adapter.setup(x => {
        x.onPost(url).replyOnce(HttpStatusCode.OK, expected)
      })

      const response = await ltcsProvisionReports.getScoreSummary({ form })

      expect(response).toStrictEqual(deleteUndefinedProperties(expected))
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onPost(url).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = ltcsProvisionReports.getScoreSummary({ form })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })
  describe('download sheet', () => {
    const url = '/api/ltcs-provision-report-sheets'
    const form: LtcsProvisionReportsApi.DownloadForm = {
      officeId: 1,
      userId: 10,
      providedIn: '2021-01',
      issuedOn: '2021-02-01',
      needsMaskingInsNumber: true,
      needsMaskingInsName: true
    }
    it('should post ltcs-provision-report-sheets', async () => {
      adapter.setup(x => {
        x.onPost(url).replyOnce(HttpStatusCode.Created)
      })

      await ltcsProvisionReports.downloadSheets({ form })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'post', url })
      expect(JSON.parse(request.data)).toStrictEqual(form)
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onPost(url).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = ltcsProvisionReports.downloadSheets({ form })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })
})
