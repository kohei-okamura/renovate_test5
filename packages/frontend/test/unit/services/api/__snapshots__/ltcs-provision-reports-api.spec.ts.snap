// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/ltcs-provision-reports-api download sheet should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/ltcs-provision-reports-api get should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/ltcs-provision-reports-api getIndex should return response of the api 1`] = `
Object {
  "list": Array [
    Object {
      "insNumber": "0141011148",
      "isEnabled": false,
      "name": Object {
        "displayName": "宮部 昴",
        "familyName": "宮部",
        "givenName": "昴",
        "phoneticDisplayName": "ミヤベ スバル",
        "phoneticFamilyName": "ミヤベ",
        "phoneticGivenName": "スバル",
      },
      "status": 2,
      "userId": 78,
    },
    Object {
      "insNumber": "2359782409",
      "isEnabled": false,
      "name": Object {
        "displayName": "木村 恵梨香",
        "familyName": "木村",
        "givenName": "恵梨香",
        "phoneticDisplayName": "キムラ エリカ",
        "phoneticFamilyName": "キムラ",
        "phoneticGivenName": "エリカ",
      },
      "status": 2,
      "userId": 71,
    },
    Object {
      "insNumber": "8051111277",
      "isEnabled": false,
      "name": Object {
        "displayName": "早川 沙知絵",
        "familyName": "早川",
        "givenName": "沙知絵",
        "phoneticDisplayName": "ハヤカワ サチエ",
        "phoneticFamilyName": "ハヤカワ",
        "phoneticGivenName": "サチエ",
      },
      "status": 3,
      "userId": 21,
    },
    Object {
      "insNumber": "7700042244",
      "isEnabled": false,
      "name": Object {
        "displayName": "浅利 惇",
        "familyName": "浅利",
        "givenName": "惇",
        "phoneticDisplayName": "アサリ アツシ",
        "phoneticFamilyName": "アサリ",
        "phoneticGivenName": "アツシ",
      },
      "status": 1,
      "userId": 67,
    },
    Object {
      "insNumber": "2267381752",
      "isEnabled": false,
      "name": Object {
        "displayName": "吉野 美帆",
        "familyName": "吉野",
        "givenName": "美帆",
        "phoneticDisplayName": "ヨシノ ミホ",
        "phoneticFamilyName": "ヨシノ",
        "phoneticGivenName": "ミホ",
      },
      "status": 3,
      "userId": 13,
    },
    Object {
      "insNumber": "2775778802",
      "isEnabled": false,
      "name": Object {
        "displayName": "松本 早紀",
        "familyName": "松本",
        "givenName": "早紀",
        "phoneticDisplayName": "マツモト サキ",
        "phoneticFamilyName": "マツモト",
        "phoneticGivenName": "サキ",
      },
      "status": 3,
      "userId": 45,
    },
    Object {
      "insNumber": "4697982719",
      "isEnabled": true,
      "name": Object {
        "displayName": "佐竹 真吾",
        "familyName": "佐竹",
        "givenName": "真吾",
        "phoneticDisplayName": "サタケ シンゴ",
        "phoneticFamilyName": "サタケ",
        "phoneticGivenName": "シンゴ",
      },
      "status": 3,
      "userId": 31,
    },
    Object {
      "insNumber": "1759064165",
      "isEnabled": false,
      "name": Object {
        "displayName": "市川 涼子",
        "familyName": "市川",
        "givenName": "涼子",
        "phoneticDisplayName": "イチカワ リョウコ",
        "phoneticFamilyName": "イチカワ",
        "phoneticGivenName": "リョウコ",
      },
      "status": 1,
      "userId": 68,
    },
    Object {
      "insNumber": "6548698044",
      "isEnabled": false,
      "name": Object {
        "displayName": "前原 栄一",
        "familyName": "前原",
        "givenName": "栄一",
        "phoneticDisplayName": "マエハラ エイイチ",
        "phoneticFamilyName": "マエハラ",
        "phoneticGivenName": "エイイチ",
      },
      "status": 1,
      "userId": 64,
    },
    Object {
      "insNumber": "7304119384",
      "isEnabled": true,
      "name": Object {
        "displayName": "木本 恵美",
        "familyName": "木本",
        "givenName": "恵美",
        "phoneticDisplayName": "キモト メグミ",
        "phoneticFamilyName": "キモト",
        "phoneticGivenName": "メグミ",
      },
      "status": 3,
      "userId": 28,
    },
  ],
  "pagination": Object {
    "count": 10,
    "desc": false,
    "itemsPerPage": 10,
    "page": 1,
    "pages": 1,
    "sortBy": "",
  },
}
`;

exports[`api/ltcs-provision-reports-api getIndex should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/ltcs-provision-reports-api ltcs-provision-report-score-summary should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/ltcs-provision-reports-api update should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/ltcs-provision-reports-api updateStatus should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;
